tests/utilTest.cpp

```diff
@@ -0,0 +1,421 @@
+    }
+    std::vector<unsigned char> v(16);
+    libzerocash::convertBytesToBytesVector(bytes, v);
+    for (int i = 0; i < 16; i++) {
+        BOOST_CHECK(v.at(i) == bytes[i]);
+    }
+}
+
+BOOST_AUTO_TEST_CASE( testConvertBytesVectorToBytes ) {
+    std::vector<unsigned char>v(16);
+    for (int i = 0; i < 16; i++) {
+        v[i] = i;
+    }
+    unsigned char bytes[16];
+    memset(bytes, 0, 16);
+    libzerocash::convertBytesVectorToBytes(v, bytes);
+    for (int i = 0; i < 16; i++) {
+        BOOST_CHECK(bytes[i] == v.at(i));
+    }
+}
+
+BOOST_AUTO_TEST_CASE( testConvertBytesVectorToVector ) {
+    std::vector<unsigned char> bytes = {0x00, 0x01, 0x03, 0x12, 0xFF};
+    std::vector<bool> expected_bits = {
+        // 0x00
+        0, 0, 0, 0, 0, 0, 0, 0,
+        // 0x01
+        0, 0, 0, 0, 0, 0, 0, 1,
+        // 0x03
+        0, 0, 0, 0, 0, 0, 1, 1,
+        // 0x12
+        0, 0, 0, 1, 0, 0, 1, 0,
+        // 0xFF
+        1, 1, 1, 1, 1, 1, 1, 1
+    };
+    std::vector<bool> actual_bits;
+    libzerocash::convertBytesVectorToVector(bytes, actual_bits);
+    BOOST_CHECK(actual_bits == expected_bits);
+}
+
+BOOST_AUTO_TEST_CASE( testConvertVectorToBytesVector ) {
+    std::vector<unsigned char> expected_bytes = {0x00, 0x01, 0x03, 0x12, 0xFF};
+    std::vector<bool> bits = {
+        // 0x00
+        0, 0, 0, 0, 0, 0, 0, 0,
+        // 0x01
+        0, 0, 0, 0, 0, 0, 0, 1,
+        // 0x03
+        0, 0, 0, 0, 0, 0, 1, 1,
+        // 0x12
+        0, 0, 0, 1, 0, 0, 1, 0,
+        // 0xFF
+        1, 1, 1, 1, 1, 1, 1, 1
+    };
+    std::vector<unsigned char> actual_bytes;
+    libzerocash::convertVectorToBytesVector(bits, actual_bytes);
+    BOOST_CHECK(actual_bytes == expected_bytes);
+}
+
+BOOST_AUTO_TEST_CASE( testConvertIntToBytesVector ) {
+    uint64_t val;
+    std::vector<unsigned char> expected;
+    std::vector<unsigned char> bytes(8);
+
+    val = 0ULL;
+    expected = { 0, 0, 0, 0, 0, 0, 0, 0 };
+    libzerocash::convertIntToBytesVector(val, bytes);
+    BOOST_CHECK( expected == bytes );
+
+    val = 1ULL;
+    expected = { 0, 0, 0, 0, 0, 0, 0, 1 };
+    libzerocash::convertIntToBytesVector(val, bytes);
+    BOOST_CHECK( expected == bytes );
+
+    val = 0xffffffffffffffffULL;
+    expected = { 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff };
+    libzerocash::convertIntToBytesVector(val, bytes);
+    BOOST_CHECK( expected == bytes );
+
+    val = 0x8000000080000001ULL; // sign extension
+    expected = { 0x80, 0x00, 0x00, 0x00, 0x80, 0x00, 0x00, 0x01 };
+    libzerocash::convertIntToBytesVector(val, bytes);
+    BOOST_CHECK( expected == bytes );
+
+    // The following two tests aren't necessarily desired results. They merely
+    // document the behavior so that we'll be alerted if it changes in the
+    // future.
+
+    val = 0xffffffffdeadbeefULL; // truncation
+    expected = { 0xde, 0xad, 0xbe, 0xef };
+    std::vector<unsigned char> small_bytes(4);
+    libzerocash::convertIntToBytesVector(val, small_bytes);
```

Comment:
Test LGTM, but as I said above, "silent" truncation is a slippery slope to security vulnerabilities.